File: 20260307/lc1.cpp
#include <cstddef>
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    vector<int> ans;
    for (size_t i = 0; i < nums.size(); i++) {
      for (size_t j = i + 1; j < nums.size(); j++) {
        if (nums[i] + nums[j] == target) {
          ans.push_back(i);
          ans.push_back(j);
          return ans;
        }
      }
    }
    return ans;
  }
};

std::vector<int> twoSum(std::vector<int> &nums, int target) {
  unordered_map<int, int> hashtable;
  for (int i = 0; i < nums.size(); i++) {
    auto it = hashtable.find(target - nums[i]);
    if (it != hashtable.end()) {
        return {it->second, i};
    }
    hashtable[nums[i]] = i;
  }
  return {};
}

int main() {
  std::vector<int> tmp = {2, 7, 11, 15};
  auto res = twoSum(tmp, 9);

  std::cout << "1st:" << res[0] << ", 2nd: " << res[1] << std::endl;
  return 0;
}

File: 20260307/lc1.js
// @ts-check

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];

    // map.has(key)：检查 key 是否存在，返回 boolean
    if (map.has(complement)) {
      // map.get(key)：取出对应的 value
      // 这里返回数组字面量，[] 是 JS 创建数组的方式
      return [map.get(complement), i];
    }
    map.set(nums[i], i);
  }
  return [];
};
